basket/news/tasks.py

```diff
@@ -256,11 +256,17 @@ def fxa_email_changed(data):
         # message older than our last update for this UID
         return
 
+    cache.set(cache_key, ts, 7200)  # 2 hr
```

Comment:
I think we should set this cache after updating everything so that if it fails and is retried that it'll try again.